### DISCUSSION TITLE
FileNotFoundException in InitializeChromium when run from command line

### DISCUSSION THREAD

**Discussion by christianwade** [cefsharp/CefSharp#4892] [General]
When I run my executable from the current folder, it works, but if I refer to it from a different folder, it doesn't (see animated GIF) - I think due to CefSharp references. It didn't do this when using an old version of CefSharp. Only since a developer upgraded to the new version for me.

![InitializeChromium error](https://github.com/user-attachments/assets/5bf93001-ba70-4e2b-ac8d-a8d731042721)

The exe is invoked by another process for which I can't control the current directory.

I got around this by setting the current directory when the process is initially invoked (see below). Does this sound like a reasonable way to deal with this, or is there a better way to deal with this?

![Set current dir](https://github.com/user-attachments/assets/654807f5-0a21-4cd8-8fed-ba88ad87977b)

**Comment by amaitland** [comment]
> It didn't do this when using an old version of CefSharp.

What version are you upgrading `from` and `to`?

Are you targeting `x86`, `x64` or `AnyCPU`? If your targeting `AnyCPU` how are you loading the arch specific libs?

> Does this sound like a reasonable way to deal with this, or is there a better way to deal with this?

From a `CefSharp` point of view, this shouldn't be a problem. Typically I'd rely on `AssemblyResolve` to load dependencies..

**Reply by christianwade** [reply]
Thanks for the clarification. ideally anycpu. The CefSharp part wasn't done by me, so I'm not sure of historical versions and whatnot. If this method to resolve works for CefSharp, I'll stick with it. Seems to be working fine.
